tests/unit/lib/intrinsic_resolver/test_intrinsic_resolver.py

```diff
@@ -0,0 +1,1424 @@
+        self.resolver = IntrinsicResolver(template=template,
+                                          symbol_resolver=symbol_resolver)
+
+    def test_basic_template_resolution(self):
+        resolved_template = self.resolver.resolve_template(ignore_errors=False)
+        expected_resources = {'HelloHandler2E4FBA4D': {'Properties': {'handler': 'main.handle'},
+                                                       'Type': 'AWS::Lambda::Function'},
+                              'LambdaFunction': {'Properties': {
+                                  'Uri': 'arn:aws:apigateway:us-east-1a:lambda:path/2015-03-31/functions/arn:aws'
+                                         ':lambda:us-east-1:406033500479:function:HelloHandler2E4FBA4D/invocations'},
+                                  'Type': 'AWS::Lambda::Function'},
+                              'RestApi': {'Properties': {'Body': 'YTtlO2Y7ZA==',
+                                                         'BodyS3Location': 'https://s3location/'},
+                                          'Type': 'AWS::ApiGateway::RestApi'},
+                              'RestApiResource': {'Properties': {'PathPart': '{proxy+}',
+                                                                 'RestApiId': 'RestApi',
+                                                                 'parentId': '/'}}}
+        self.assertEquals(resolved_template, expected_resources)
+
+    def test_template_fail_errors(self):
+        resources = deepcopy(self.resources)
+        resources["RestApi.Deployment"]["Properties"]["BodyS3Location"] = {"Fn::FindInMap": []}
+        template = {
+            "Mappings": self.mappings,
+            "Conditions": self.conditions,
+            "Resources": resources,
+        }
+        symbol_resolver = IntrinsicsSymbolTable(template=template,
+                                                logical_id_translator=self.logical_id_translator)
+        resolver = IntrinsicResolver(template=template,
+                                     symbol_resolver=symbol_resolver)
+        with self.assertRaises(InvalidIntrinsicException, msg="Invalid Find In Map"):
+            resolver.resolve_template(ignore_errors=False)
+
+    def test_template_ignore_errors(self):
+        resources = deepcopy(self.resources)
+        resources["RestApi.Deployment"]["Properties"]["BodyS3Location"] = {"Fn::FindInMap": []}
+        template = {
+            "Mappings": self.mappings,
+            "Conditions": self.conditions,
+            "Resources": resources,
+        }
+        symbol_resolver = IntrinsicsSymbolTable(template=template,
+                                                logical_id_translator=self.logical_id_translator)
+        resolver = IntrinsicResolver(template=template,
+                                     symbol_resolver=symbol_resolver)
+        result = resolver.resolve_template(ignore_errors=True)
+        expected_template = {'HelloHandler2E4FBA4D': {'Properties': {'handler': 'main.handle'},
```

Comment:
can the larger templates be added as separate files, that can be checked against. maybe under testdata/?